java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -430,6 +430,70 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_dropListDuplicates(JNIEnv
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_dropListDuplicatesWithKeysValues(
+    JNIEnv *env, jclass, jlong keys_vals_handle) {
+  JNI_NULL_CHECK(env, keys_vals_handle, "keys_vals_handle is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+    auto const *input_cv = reinterpret_cast<cudf::column_view const *>(keys_vals_handle);
+    CUDF_EXPECTS(input_cv->offset() == 0, "Input column has non-zero offset.");
+    CUDF_EXPECTS(input_cv->type().id() == cudf::type_id::LIST,
+                 "Input column is not a lists column.");
+
+    // Extract list offsets and a column of struct<keys, values> from the input lists column.
+    auto const lists_keys_vals = cudf::lists_column_view(*input_cv);
+    auto const keys_vals = lists_keys_vals.child();
```

Comment:
The input column could be sliced, no?
```suggestion
    auto const keys_vals = lists_keys_vals.get_sliced_child();
```

Reply:
We can't know. Sliced to be safe.

Reply:
Technically the java code never slices without copying the input back out, because we have a lot of code that cannot handle sliced columns.  But it is better to be safe and not add more to that tech debt.